File: leet_226.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        
        #Checks if tree is empty
        if root == None:
            return root

        #Stores left and right child in a temporary variable
        left_temp = root.left
        right_temp = root.right

        #Swap right and left child
        root.right = left_temp
        root.left = right_temp

        #Invert SubTree
        self.invertTree(left_temp)
        self.invertTree(right_temp)
        
        return root

File: leet_226.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == nullptr)
            return root;
        
        //Store left and right child in temporary variable
        TreeNode* left_child = root ->left;
        TreeNode* right_child = root -> right;

        root->right = left_child;
        root->left = right_child;

        invertTree(left_child);
        invertTree(right_child);

        return root;

    }
};